### ISSUE TITLE
Please add Pressure support

### ISSUE THREAD

**Issue by x2harry** [thorrak/fermentrack#466]
Hello
I brew under pressure, will there be a pressure sensor added to Fermentrack like in brewpiless?
Keeping the fermentation vessel at a set pressure and operating a valve to stop it going over pressure.
Cheers and thanks for all your good work.

**Comment by FordMustang0288** [comment]
Funny that I thought I was onto something unique today. Guess not haha. 

I am interested in this even though I only have a fermenter that can handle 3 PSI. The reason is I feel like it is extremely hard to add < 3 psi from a tank. At the very least, monitoring my PRV would be great. But the addition of a solenoid on both input/output would be even better!

https://www.amazon.com/Pressure-Transducer-Sender-Sensor-Stainless/dp/B0748BXFYQ/ref=sr_1_56_sspa?dchild=1&keywords=wireless+pressure+sensor+npt&qid=1595441056&sr=8-56-spons&psc=1&spLa=ZW5jcnlwdGVkUXVhbGlmaWVyPUEyTU1JVkZNMVRSNDNOJmVuY3J5cHRlZElkPUEwNDQxOTk4Q0xNQ09WWDBWQk9XJmVuY3J5cHRlZEFkSWQ9QTAwNjY5ODgxRkxaOFBOUThQOFI1JndpZGdldE5hbWU9c3BfYnRmJmFjdGlvbj1jbGlja1JlZGlyZWN0JmRvTm90TG9nQ2xpY2s9dHJ1ZQ==

https://www.instructables.com/id/MyBrewbot-Operating-Instructions/

http://www.lowoxygenbrewing.com/brewing-methods/spunding-automation-and-logic/

**Comment by lbussy** [comment]
IANT (I am not @thorrak) but this seems like a solution in search of a problem.

1. Why would you want an electronic solenoid when the defacto standard/exemplar of a diaphragm pressure regulator works so well?  Why not just make the same spunding valve everyone else uses?
2. The ESP8266 is out of pins.  You won't be getting any additional I/O out of that platform.  Anything additional would need to wait for a port to the ESP32.

I could see monitoring pressure, but you would still be waiting for a port to the ESP32 in order to add the inputs.

**Comment by mp-se** [comment]
I've started to do pressure fermentation myself and wanted to measure the pressure so I built a device around the esp8266.  This could be added to the brewpi controller once it's ported to ESP32. I just wanted to skip the step running down into the basement to figure out what the current pressure was. 

I've got something up and running that I'm currently testing on my latest brew. If you want have a look and give me some new ideas the project can be found here; https://github.com/mp-se/pressuremon

**Comment by Usernametakenasdf** [comment]
> IANT (I am not @thorrak) but this seems like a solution in search of a problem.
> 
> 1. Why would you want an electronic solenoid when the defacto standard/exemplar of a diaphragm pressure regulator works so well?  Why not just make the same spunding valve everyone else uses?
> 2. The ESP8266 is out of pins.  You won't be getting any additional I/O out of that platform.  Anything additional would need to wait for a port to the ESP32.
> 
> I could see monitoring pressure, but you would still be waiting for a port to the ESP32 in order to add the inputs.

In answer to question #1:
1. I used to use brewpiless that had this functionality along with fermentation profiles triggered from SG from my iSpindel. I also used to work away for 2 weeks out of every 3. I could pitch my yeast before I left, let it ferment whilst I was away and come back to fermented, cold crashed and carbonated beer without me having to touch a thing. If it didn't lose wifi connectivity or whatever other instability the code had that week.

2. I'm a massive nerd and it's very cool.



People have lots of reasons to do/want things. You might not see the benefit, others very much do.

**Comment by lbussy** [comment]
I've been brewing a couple of years, and I can come up with limited reasons to need to vary the pressure for spunding.  To each their own, I guess.

As I said, no pins, so this will have to be after the ESP32 port is done.

**Comment by mp-se** [comment]
There is no support for handling pressure in fermentrack yet. I have bulilt a device that can measure the pressur and I currently send the data to fermentrack emulating an ispindle, so this way i can get a graph over how the pressure varies during fermentation. But i would agree with lbussy that seeing the pressure is of limited use since i have a spunding valve that releases the pressure over a certain level. It can probably be used for seeing how active the fermentation is for example.

**Comment by lbussy** [comment]
I don't have a personal issue with it, but it feels like a failure point that doesn't have to be.  Monitoring a critical mechanical process seems safer than digital control.  A power outage would mean you have a mess (or worse) waiting to happen.

**Comment by thorrak** [comment]
Yeah -- the biggest issue that I see is someone ignoring any advice to always have an automatic PRV that releases over a specific pressure limit in addition to a digitally-controlled PRV and accidentally blowing up their garage/closet/etc.

That said - no issues whatsoever with implementing pressure _monitoring_ other than the concern you noted about a lack of pins (and, of course, getting the actual code written). It's not something I'm likely to personally implement absent acquiring a pressure sensor I wanted to play with, but it's one that I'd entertain merging if someone drafted a PR.

**Comment by mp-se** [comment]
I was considering doing some work on that, but creating a new pressure data type in django is over my skill level at the moment. Getting the data in is probably a part i could cope with since i could use the ispindle as a template…
